src/main/resources/public/js/graphs.controller.js

```diff
@@ -109,22 +109,22 @@
             groupByKey = label + DOT + readableDateShort[0];
           }
         }
-        var distance = parseFloat(activities[i].distance.toFixed(2, 2));
+        const distance = parseFloat(sortedActivities[i].distance.toFixed(2, 2));
```

Comment:
@sepe81 After all tests, the real problem was, that in this line an line 117 sortedActivities was missing. Time to go to bed. ;)